lib/github/kv.rb

```diff
@@ -252,6 +253,32 @@ def setnx(key, value, expires: nil)
       }
     end
 
+    # increment :: String, Integer, expires: Time? -> Integer
+    #
+    # Increment the key's value by an amount.
+    #
+    # key     - The key to increment.
+    # amount  - The amount to increment the key's value by.
+    # expires - When the key should expire.
+    #
+    # Returns the key's value after incrementing.
+    def increment(key, amount: 1, expires: nil)
+      sql = GitHub::SQL.run(<<-SQL, key: key, amount: amount, now: now, expires: expires || GitHub::SQL::NULL, connection: connection)
+        INSERT INTO key_values (`key`, value, created_at, updated_at, expires_at)
+        VALUES(:key, :amount, :now, :now, :expires)
+        ON DUPLICATE KEY UPDATE
+          `value`=LAST_INSERT_ID(IF(
+            `expires_at` IS NULL OR `expires_at`>=NOW(),
+            `value`+:amount,
+            :amount
+          )),
+          `updated_at`=:now,
+          `expires_at`=:expires
+      SQL
+
+      sql.affected_rows == 1 ? amount : sql.last_insert_id
```

Comment:
As per https://dev.mysql.com/doc/refman/8.0/en/insert-on-duplicate.html 

> With ON DUPLICATE KEY UPDATE, the affected-rows value per row is 1 if the row is inserted as a new row, 2 if an existing row is updated, and 0 if an existing row is set to its current values.

`sql.affected_rows` will return 1 if an insert was performed. In this case we know the key's value is set to the amount so we can just return the amount directly otherwise we return the `LAST_INSERT_ID` which we set to be the new value.